generator/src/main/java/org/corfudb/generator/distributions/DataSet.java

```diff
@@ -0,0 +1,51 @@
+package org.corfudb.generator.distributions;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ *
+ * This class defines a generic data distribution. A DataSet is used by operations
+ * that require data to execute (i.e. writing).
+ *
+ * Created by maithem on 7/14/17.
+ */
+public abstract class DataSet {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Convert the abstract class "DataSet" into an interface. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1610)